### PR TITLE
Improve OAuth related logging

### PR DESCRIPTION
**Pull request by fmoc**
This commit also improves the code structure a little.

### CODE REVIEW

File: src/libsync/creds/oauth.cpp [hunk 1]
```diff
@@ -249,32 +261,43 @@ void OAuth::startAuthentication()
...
             });
-            connect(job, &RegisterClientJob::errorOccured, this, [this](const QString &error) {
+            connect(registerJob, &RegisterClientJob::errorOccured, this, [this](const QString &error) {
                 qCWarning(lcOauth) << "Failed to dynamically register the client, try the default client id" << error;
                 Q_EMIT authorisationLinkChanged();
             });
-            job->start();
+            registerJob->start();
         } else {
+            qCDebug(lcOauth) << "dynamic registration disabled and/or registration endpoint invalid";
```
Comment (TheOneRing) [inline src/libsync/creds/oauth.cpp:278]: hmm better log depending on the case.
Also its not invalid but not present.

File: src/libsync/creds/oauth.cpp [hunk 2]
```diff
@@ -457,17 +480,25 @@ void OAuth::fetchWellKnown()
         OC_ASSERT(!urls.second.isNull());
         _authEndpoint = QUrl::fromUserInput(urls.first);
         _tokenEndpoint = QUrl::fromUserInput(urls.second);
+
+        qCDebug(lcOauth) << "override URL set, using auth endpoint" << _authEndpoint << "and token endpoint" << _tokenEndpoint;
```
Comment (TheOneRing) [inline src/libsync/creds/oauth.cpp:484]: hmm not an override its more that the other values are the fallback

File: src/libsync/creds/oauth.cpp [hunk 3]
```diff
@@ -576,12 +620,19 @@ void AccountBasedOAuth::fetchWellKnown()
...
     }
+
     _isRefreshingToken = true;
+
+    qCDebug(lcOauth) << "fetching dynamic registration data";
+
     auto credentialsJob = _account->credentialManager()->get(dynamicRegistrationDataC());
 
-    connect(credentialsJob, &CredentialJob::finished, this, [=] {
+    connect(credentialsJob, &CredentialJob::finished, this, [this, credentialsJob, refreshToken] {
```
Comment (TheOneRing) [inline src/libsync/creds/oauth.cpp:633]: I prefer, to put this last, but we should come up with a standard on how to capture this.

File: src/libsync/creds/oauth.cpp [hunk 4]
```diff
@@ -629,20 +680,23 @@ void AccountBasedOAuth::refreshAuthentication(const QString &refreshToken)
...
                     refresh();
                 });
-                connect(registerJob, &RegisterClientJob::errorOccured, this, [refresh, this](const QString &error) {
+                connect(registerJob, &RegisterClientJob::errorOccured, this, [refresh](const QString &error) {
                     qCWarning(lcOauth) << "Failed to dynamically register the client, try the default client id" << error;
                     refresh();
                 });
                 registerJob->start();
             } else {
+                qCDebug(lcOauth) << "dynamic registration disabled and/or registration endpoint invalid";
```
Comment (TheOneRing) [inline src/libsync/creds/oauth.cpp:699]: see ^

File: src/libsync/creds/oauth.cpp (hunk 1 above)
Comment (fmoc) [reply src/libsync/creds/oauth.cpp:278]: This is a direct reference to the `if`'s `isValid` call. I guess I'll add a docstring to clarify the intention of this call.

File: src/libsync/creds/oauth.cpp (hunk 2 above)
Comment (fmoc) [reply src/libsync/creds/oauth.cpp:484]: I'm not sure I understand you right. This `if` handles the case when the theme enforces some URLs.

Comment (TheOneRing) [reply src/libsync/creds/oauth.cpp:484]: Sorry parsed that wrong

File: src/libsync/creds/oauth.cpp [hunk 5]
```diff
@@ -695,10 +707,12 @@ void AccountBasedOAuth::refreshAuthentication(const QString &refreshToken)
                     refresh();
                 });
                 registerJob->start();
-            } else {
-                qCDebug(lcOauth) << "dynamic registration disabled and/or registration endpoint invalid";
-                refresh();
```
Comment (TheOneRing) [inline src/libsync/creds/oauth.cpp:646]: What happened to the refresh?

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/libsync/creds/oauth.cpp (hunk 5 above)
Comment (fmoc) [reply src/libsync/creds/oauth.cpp:646]: Copy pasta mistake. Fixed.

**Review by TheOneRing — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13465/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13465/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13465/guiReportUpload/serverlog.log

Screenshots:
- [share_a_file_with_many_users.png](https://cache.owncloud.com/public/owncloud/client/13465/guiReportUpload/screenshots/share_a_file_with_many_users.png)

</details>

**Pull request merged**